extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -536,37 +574,152 @@ export class AmpStoryEmbeddedComponent {
   }
 
   /**
-   * Animates into expanded view.
+   * Returns expanded element back to original state.
+   * @private
+   */
+  closeExpandedEl_() {
+    this.triggeringTarget_.classList.toggle(
+        'i-amphtml-expanded-component', false);
+    const embedId =
+      this.triggeringTarget_.getAttribute(EMBED_ID_ATTRIBUTE_NAME);
+
+    const styleEl = embedStyleEls[embedId];
+
+    const transformString = styleEl.textContent.substring(
```

Comment:
Sorry, I'm not sure I get the whole scope of the PR here, but can't you get this our of the object representation instead of needing to rely on string parsing?

Reply:
I'm extracting the part of the style element's rule which corresponds to the transform, so I can replace that with the new transform string. (See line 593 below, where this `transformString` var is used).

Reply:
Is it possible to regenerate the entire style string instead, and then just reset the whole `textContent`?  Or am I missing something?

Reply:
This look very easy to break. :( Didn't you create `AMP_EMBED_DATA` to avoid working on the `textContent` string?

Reply:
As discussed offline we need to keep the `<style>` and `AMP_EMBED_DATA` out of sync because we want to store a `scale` / `translate` to get back to after the embed has been expanded. But I introduced a helper method that should make this clearer.

Reply:
Sure, just thought it would be easier to read this way. I made the change.